includes/class-wc-post-types.php

```diff
@@ -29,6 +29,7 @@ public static function init() {
 		add_action( 'init', array( __CLASS__, 'register_post_status' ), 9 );
 		add_action( 'init', array( __CLASS__, 'support_jetpack_omnisearch' ) );
 		add_filter( 'rest_api_allowed_post_types', array( __CLASS__, 'rest_api_allowed_post_types' ) );
+		add_action( 'woocommerce_flush_rewrite_rules', array( __CLASS__, 'flush_rewrite_rules' ) );
```

Comment:
No need to call a class method, you can use simply `flush_rewrite_rules ` as the callback.